wandb/apis/public.py

```diff
@@ -2085,18 +2090,27 @@ def versions(self, per_page=50):
     def __repr__(self):
         return "<ArtifactCollection {} ({})>".format(self.name, self.type)
 
+
 class Artifact(object):
 
     def __init__(self, client, entity, project, name, attrs=None):
         self.client = client
         self.entity = entity
         self.project = project
         self.artifact_name = name
-        self._collection_name = None
         self._attrs = attrs
-        self._metadata = {}
         if self._attrs is None:
             self._load()
+        self._metadata = self._attrs.get("metadata", None)
+        self._description = self._attrs.get("description", None)
+        self._sequence_name = self._attrs["artifactSequence"]["name"]
+        self._version_index = self._attrs["versionIndex"]
```

Comment:
Should we default this to 0 if we don't have a `versionIndex`

Reply:
Because of the way the API works, you can only resolve artifacts that have been committed and consequently have a `versionIndex`, however a default might still be good here.

Reply:
Fixed.